Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -130,16 +130,31 @@ extension NSManagedObjectContext: AttachmentDatabaseSession {
         dto.type = type
         dto.title = seed.fileName
         
-        let attachment = ChatMessageDefaultAttachment(
-            id: id,
-            type: AttachmentType(rawValue: type),
-            localURL: seed.localURL,
-            localState: dto.localState,
-            title: seed.fileName,
-            file: seed.file
-        )
-        
-        dto.data = try JSONEncoder.stream.encode(attachment)
+        if isAttachmentModelSeparationChangesApplied {
+            var attachment: Encodable
+            
+            switch seed.type {
+            case .image:
+                attachment = ChatMessageImageAttachment(title: seed.fileName)
+            case .file:
+                attachment = ChatMessageFileAttachment(title: seed.fileName, file: seed.file)
+            default:
+                throw ClientError.AttachmentSeedUploading(id: id)
```

Comment:
This will go away, right?

Reply:
I'm not sure. 
We should somehow throw an error when `Seed` is created with the type that we don't know how to upload.